source/Mlos.Python/mlos/Optimizers/RegressionModels/RegressionEnhancedRandomForestModel.py

```diff
@@ -149,6 +140,14 @@ def __init__(
 
         # one hot encode categorical input dimensions
         self.one_hot_encoder_adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=input_space, merge_all_categorical_dimensions=True, drop='first')
+        # Explode continuous dimensions to polynomial features up to model config specified monomial degree
+        # am using include_bias to produce constant term (all 1s) column to simplify one hot encoding logic
+        self.polynomial_features_adapter = ContinuousToPolynomialBasisHypergridAdapter(
+            adaptee=self.one_hot_encoder_adapter.target,

```

Comment:
I think it would be better to pass the entire `self.one_hot_encoder_adapter` as an adaptee. But I suspect you might have had a reason for not doing so, so I will keep reading :)

Reply:
```suggestion
            adaptee=self.one_hot_encoder_adapter,
```